crates/ide-diagnostics/src/handlers/incorrect_case.rs

```diff
@@ -344,24 +345,6 @@ fn main() {
         );
     }
 
-    #[test]
-    fn non_let_bind() {
```

Comment:
shouldnt this test stay?

Reply:
I ended up replacing it with [this test](https://github.com/rust-lang/rust-analyzer/pull/17588/files/af30111b1c8af7abdaaa52120e24e088f15ff25b#diff-4fb2429f42867a6c75f039b611c8086c035d6daf099dbdaffb5e6fd9a78927c5R817) inside of `test_match` below:
```rust
enum Foo { V1, V2 }
use Foo::V1;
fn main() {
    match V1 {
        _Bad1 @ V1 => {},
     // ^^^^^ 💡 warn: Variable `_Bad1` should have snake_case name, e.g. `_bad1`
        Foo::V2 => {}
    }
}
```
I did this to group the `match` tests, and because the old test has an issue: it doesn't `use Option::None`, so the `None` in `SOME_VAR @ None` actually referes to a new variable binding, rather than an enum variant.
This is the same issue I fixed in `missing_match_arms.rs` [here](https://github.com/rust-lang/rust-analyzer/pull/17588/files/af30111b1c8af7abdaaa52120e24e088f15ff25b#diff-332f11ebf6f6a14d806c673271b14832507641612034ccbec801318bafff5005R611)

Reply:
Gotcha